jobs/cloud-provider/spec

```diff
@@ -5,17 +5,39 @@ templates:
   config/cloud-provider.ini.erb: config/cloud-provider.ini
   bin/cloud-provider_utils.erb: bin/cloud-provider_utils
 
-
 packages:
 - pid_utils
 
 properties:
-  cloud-provider:
-    type:
-      description: "Type of Cloud Provider to use"
-      example: gce
-    gce:
-      project-id:
-        description: Google Cloud project id. Required when cloud-provider.type is gce
-      network-name:
-        description: Google Cloud network name. Required when cloud-provider.type is gce
+  cloud-provider.type:
+    description: "Type of Cloud Provider to use"
+    example: gce, vsphere
+  cloud-provider.gce.project-id:
+    description: Google Cloud project id. Required when cloud-provider.type is gce
+  cloud-provider.gce.network-name:
+    description: Google Cloud network name. Required when cloud-provider.type is gce
+  cloud-provider.vsphere.user:
```

Comment:
This is a nit pick, but can you add some spacing and comments above the vsphere and gce fields? Just to make it more obvious that they are different sections.